### PR TITLE
Feature: First iteration of the 'flavours' feature

### PR DESCRIPTION
**Pull request by abias**
This pull request adds the first iteration of the 'flavours' feature. It strives to solve #25 (and solves 3 of the 4 items which were raised there indeed), but it does much more: It introduces a basis for changing the look and feel of particular Moodle pages based on particular conditions.

### CODE REVIEW

**Comment by abias** [comment]
Up to now, this PR is still a draft. There are two pieces of work remaining:
1. Add a caching layer to avoid as much additional database queries as possible per page load
2. Add Behat tests.

File: classes/output/core_renderer.php [hunk 1]
```diff
@@ -39,14 +39,63 @@ class core_renderer extends \theme_boost\output\core_renderer {
      * Returns the moodle_url for the favicon.
      *
      * This renderer function is copied and modified from /lib/outputrenderers.php
-     * It uses the same logic as Moodle 4.1dev already which introduced a Moodle core favicon setting,
+     *
+     * It checks if the favicon is overridden in a flavour and, if yes, it serves this favicon.
+     * If there isn't a favicon in any flavour set, it continues with the logic from Moodle core.
+
+     * Doing this, it uses the same logic as Moodle 4.1dev already which introduced a Moodle core favicon setting,
```
Comment (lucaboesch) [inline classes/output/core_renderer.php:46]: It's not 4.1dev any more, …

Comment (abias) [reply classes/output/core_renderer.php]: I corrected this and pushed an updated patch.

File: .github/workflows/moodle-plugin-ci.yml [hunk 2]
```diff
@@ -29,9 +29,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        php: ['7.3', '7.4', '8.0']
+        php: ['7.4']
```
Comment (lucaboesch) [inline .github/workflows/moodle-plugin-ci.yml:32]: Dear @abias why would you drop PHP 8.0 testing?

Comment (lucaboesch) [reply .github/workflows/moodle-plugin-ci.yml:32]: I guess for fater testlng, then you would add it back in again, right?

Comment (abias) [reply .github/workflows/moodle-plugin-ci.yml:32]: Exactly. I am currently working on the last tests and then this is ready for a final review.

**Comment by abias** [comment]
I think / hope the code is now ready for a final review.
@lucaboesch - I am aware that it's quite a big amount of new code, but may I ask you to have a look at it?

**Review by lucaboesch — DISMISSED** [review]
Dear @abias your contributed code works and is very impressive.
I am very confident that this will give Boost Union users a real good base to work on and it is an extremely valuable contributions.
Really, I didn't find anything worrying and really worth mentioning in your code with the exception of the two small comments I've made (of which one is concerning a comment, and therefore really not visible to users, and the other is a personal taste question).
Feel free to merge this pull request with or without amending the two spots I mentioned.

2023 starts fabulously.

Best,
Luca

**Comment by abias** [comment]
Hi @lucaboesch ,

thank you very much for your kind review!
I just fixed the two small things which you mentioned in your comments and have rebased the PR onto the latest master branch.

I will merge it then as soon as the tests have passed again.

Cheers,
Alex

**Pull request merged**